.github/workflows/build_and_test.yml

```
@@ -0,0 +1,287 @@
+            .gradle/caches
+            gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Get gradle wrapper and build dependencies
+        run: |
+          if [ ! -f gradle/wrapper/gradle-wrapper.jar ]; then
+            ./configure.sh
+            ./gradlew --no-daemon dependencies
+          fi
+
+      - name: Run tests
+        run: |
+          ./gradlew --no-daemon --stacktrace test
+
+  integration-tests:
+    needs: prepare-rskj
+    runs-on: ubuntu-latest
+    container:
+      image: openjdk:8-jdk
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Setup System Tools
+        run: |
+          apt update -y
+          apt install -y curl
+
+      - uses: actions/cache@v4
+        name: Restore gradle cache
+        id: cache-gradle
+        with:
+          path: |
+            .gradle/caches
+            gradle/wrapper
+          key: ${{ runner.os }}-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Get gradle wrapper and build dependencies
+        run: |
+          if [ ! -f gradle/wrapper/gradle-wrapper.jar ]; then
+            ./configure.sh
+            ./gradlew --no-daemon dependencies
+          fi
+
+      - name: Run tests
+        run: |
+          ./gradlew --no-daemon --stacktrace integrationTest
+
+  mining-tests:
+    needs: prepare-rskj
+    runs-on: ubuntu-latest
+    services:
+      bitcoind1:
+        image: ghcr.io/rsksmart/rskj/mit_bitcoind1:latest
+        ports:
+          - 8331:8331
+          - 31591:31591
+          - 32591:32591
+        options: --name bitcoind1
+      bitcoind2:
+        image: ghcr.io/rsksmart/rskj/mit_bitcoind2:latest
+        ports:
+          - 8332:8332
+          - 31592:31592
+          - 32592:32592
+        options: --name bitcoind2
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        uses: actions/setup-node@v4
+        with:
+          node-version: '12.x'
+      - name: Check Node.js version
+        run: node --version
+
+      - name: Checkout Mining Integration Tests Repository
+        uses: actions/checkout@v4
+        with:
+          repository: rsksmart/mining-integration-tests
+          ref: ${{ secrets.MINING_INTEGRATION_TESTS_REF }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+          path: mining-integration-tests
+
+      - name: Install Mining Integration Tests dependencies
+        working-directory: mining-integration-tests
+        run: |
+          npm ci
+
+      - name: Change mining-integration-tests bitcoind url in config.json to localhost
+        working-directory: mining-integration-tests
+        run: |
+          jq '.bitcoind.url = "localhost"' config.json > config.json.tmp && mv config.json.tmp config.json
```

This construction is error prone. I prefer the following, way more complex expression:

```suggestion
          jq '.bitcoind? | if has("url") then (.url = "localhost") else error(".bitcoind.url not found") end' config.json > config.json.tmp && mv config.json.tmp config.json
```

The reason is that your proposal will create the keys in the object if they don't exist. My proposal will blow up right here if the format of `config.json` ends up being modified, making it easier to detect the issue.

---

done